lightningd/invoice.c

```diff
@@ -1002,8 +1002,14 @@ static struct command_result *json_invoice(struct command *cmd,
 	info->b11->description_hash = NULL;
 	info->b11->payment_secret = tal_dup(info->b11, struct secret,
 					    &payment_secret);
-	info->b11->features = get_offered_bolt11features(info->b11);
 
+	/* Which features should we announce to the node receiving this invoice?
+	 * This is a combination of natively supported features and featurebits
+	 * that plugins asked us to include in the invoice. */
+	info->b11->features = featurebits_or(
+	    info->b11, take(get_offered_bolt11features(info->b11)),
```

Comment:
This confused me: the standard take() pattern is to allocate of NULL, i.e. `take(get_offered_bolt11features(NULL))`.  This gets detected as a leak if something doesn't get taken, and is also clear what's meant to happen ("I'm allocating this but deliberately handing it off to get taken immediately").